src/foamadapter/solver/pimplefoam.py

```diff
@@ -0,0 +1,144 @@
+
+import pybFoam
+from pybFoam import (
+    Info,
+    Time,
+    Word,
+    adjustPhi,
+    computeCFLNumber,
+    constrainHbyA,
+    constrainPressure,
+    createPhi,
+    dictionary,
+    fvc,
+    fvm,
+    fvMesh,
+    fvScalarMatrix,
+    fvVectorMatrix,
+    pimpleControl,
+    setRefCell,
+    solve,
+    surfaceScalarField,
+    volScalarField,
+    volVectorField,
+)
+from pybFoam.turbulence import incompressibleTurbulenceModel, singlePhaseTransportModel
+
+
+class CFLNumber:
+
+    def __init__(self):
+        self.criteria = []
+        self.GREAT = 1e30
+        self.SMALL = 1e-15
+        self.maxDeltaT = 1e-3  # Default maximum deltaT
+
+    def setDelta(self, runTime, phi):
+        deltaT = runTime.deltaTValue()
+
+        if not self.criteria:
+            return
+
+        max_cfl_number, mean_cfl_number = computeCFLNumber(phi)
+        Info(f"Courant Number mean: {mean_cfl_number}, max: {max_cfl_number}")
+        ratios = [self.max_cfl_number / max_cfl_number]
+
+        # limit to 1.2 to avoid too large time steps
+        ratios = [min(ratio, 1.2) for ratio in ratios if ratio > self.SMALL and ratio < self.GREAT]
+
+        # Set most restrictive time step
+        finalDeltaT = min(min(deltaT * ratio for ratio in ratios), self.maxDeltaT)
+        runTime.setDeltaT(finalDeltaT)
+        Info(f"deltaT = {runTime.deltaTValue()}")
+        runTime.increment()
+
+
+def create_fields(mesh):
+    p = volScalarField.read_field(mesh, "p")
+    U = volVectorField.read_field(mesh, "U")
+    phi = createPhi(U)
+
+    laminarTransport = singlePhaseTransportModel(U, phi)
+
+    turbulence = incompressibleTurbulenceModel.New(U, phi, laminarTransport)
+
+    return p, U, phi, laminarTransport, turbulence
+
+
+class PimpleFoam:
+    def __init__(self, argv):
+        self._argv = argv
+        self.pRefCell = None
+        self.pRefValue = None
```

Comment:
Could be moved out of init if it is a default value and no way to modify it in __init__
```suggestion
class PimpleFoam:
    pRefCell = None
    pRefValue = None 
        
    def __init__(self, argv):
        self._argv = argv

```

Reply:
In the future there would be more instance of PimpleFoam. So, a class method would be not ideal